### PR TITLE
Fix Library bulk-selection hit target, focus ring, and Select All label

### PR DESCRIPTION
**Pull request by moona3k**
## What & why

Three fixes to the Library/Meetings **Select Many** bulk-selection UI, reported from real use, plus a label cleanup.

### 1. Clicking the selection circle did nothing
On a Library thumbnail card, clicking the rest of the card toggled selection but clicking **directly on the circle** did nothing. The circle (`selectionBadge`) is an `.overlay` layered *on top of* the card `Button` and was hit-testable, so taps landing on it were swallowed instead of reaching the Button's toggle.
**Fix:** `.allowsHitTesting(false)` on the decorative badge so the whole card surface — circle included — routes to the Button. (`MeetingRowCard` puts its badge *inside* the Button, which is exactly why only the thumbnail grid had the bug.)

### 2. Blue line + jank entering selection mode
Entering selection mode flashed a blue line across the top of the content (only the first time) and felt janky. The content is `.focusable(isBulkSelectionModeEnabled)` and gets programmatically focused to capture ⌘A / Delete; macOS draws the **system focus ring** (system-accent *blue*, not the app's coral) on first-responder acquisition — that's the line, and its first-frame draw is the felt hitch.
**Fix:** `.focusEffectDisabled()` on `TranscriptionLibraryView` and `MeetingsView` — keeps focus + key handling, removes the ring and its draw cost. (macOS 14+, matches our deployment target.)

### 3. "Select Visible" → "Select All"
The bulk action was conditionally labeled `Select Visible` (paginated) / `Select All` (fully loaded). Renamed to an unconditional **Select All**, matching the Dictation surface and Finder-style "select what's loaded" semantics. It still targets only loaded rows *by design*, so deletion never reaches unloaded records; the selected-count chip and delete confirmation always state exact counts. Drops the now-constant `selectVisibleTitle` parameter. Specs updated.

## Known follow-up (intentionally not in this PR)
The broad container `.animation(_:value:)` still reflows the lazy grid and inserts every card's badge in one pass on mode entry — a separate, *measurable* smoothness improvement. Left out to avoid a speculative layout change; the focus-ring removal was the high-impact, low-risk half.

## Verification
- `swift build` clean; full `swift test` green.
- Two independent fresh-eye reviews (Swift/SwiftUI reviewer + a second-opinion model) converged on **ship**, confirming both modifiers are correct and side-effect-free on macOS 14 (focus/⌘A/Delete and VoiceOver preserved).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Fixes the Library/Meetings Select Many UX: the selection circle now toggles as expected, the blue focus-ring flash is gone, and the action is standardized to “Select All” (still scoped to loaded rows) and disabled when no rows are visible.

- **Bug Fixes**
  - Make the thumbnail selection badge non-hit-testable so clicks on the circle toggle selection.
  - Suppress the macOS focus ring on selection mode entry while keeping ⌘A/Delete handling.
  - Always show “Select All”; remove `selectVisibleTitle`, keep scope to loaded rows, and update specs.
  - Treat zero visible rows as “all selected” to disable Select All; add a test for empty-search results.

<sup>Written for commit f295ff82090d1c50506ff1f777b00cfccaa79581. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/moona3k/macparakeet/pull/594?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: Sources/MacParakeet/Views/Transcription/BulkTranscriptionSelectionBar.swift [hunk 1]
```diff
@@ -146,9 +145,13 @@ struct BulkTranscriptionSelectionBar: View {
...
+    // (and therefore deletion) never reaches rows that aren't loaded yet. The
+    // selected-count chip and the delete confirmation always state the exact
+    // number, so the scope is explicit at the moment of action.
     private var selectVisibleAction: some View {
         SelectionBarActionButton(
-            title: selectVisibleTitle,
+            title: "Select All",
             systemImage: "checkmark.circle",
             tone: .utility,
             isDisabled: areAllVisibleSelected || isPerformingOperation,
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/MacParakeet/Views/Transcription/BulkTranscriptionSelectionBar.swift:157]: Fixed in f295ff820: `areAllLoadedVisibleTranscriptionsSelected` now treats an empty loaded-visible set as selected, which disables Select All when a search has no matches. Added `testAllLoadedVisibleTranscriptionsSelectedWhenSearchHasNoMatches`; focused `TranscriptionLibraryViewModelTests` passes (29 tests, 0 failures).
